src/Bicep.LangServer/Deploy/DeploymentHelper.cs

```diff
@@ -0,0 +1,114 @@
+            {
+                return LangServerResources.MissingLocationDeploymentFailedMessage;
+            }
+
+            DeploymentCollection? deploymentCollection;
+            var resourceIdentifier = new ResourceIdentifier(id);
+
+            if (scope == LanguageConstants.TargetScopeTypeResourceGroup ||
+                scope == LanguageConstants.TargetScopeTypeSubscription ||
+                scope == LanguageConstants.TargetScopeTypeManagementGroup)
+            {
+                try
+                {
+                    deploymentCollection = deploymentCollectionProvider.GetDeploymentCollection(armClient, resourceIdentifier, scope);
+                }
+                catch (Exception e)
+                {
+                    return string.Format(LangServerResources.DeploymentFailedWithExceptionMessage, e.Message);
+                }
+            }
+            else
+            {
+                return LangServerResources.UnsupportedScopeDeploymentFailedMessage;
+            }
+
+            if (deploymentCollection is not null)
+            {
+                JsonElement parameters;
+
+                if (string.IsNullOrWhiteSpace(parameterFilePath))
+                {
+                    parameters = JsonDocument.Parse("{}").RootElement;
+                }
+                else
+                {
+                    try
+                    {
+                        string text = File.ReadAllText(parameterFilePath);
+                        parameters = JsonDocument.Parse(text).RootElement;
+                    }
+                    catch(Exception e)
+                    {
+                        return string.Format(LangServerResources.InvalidParameterFileDeploymentFailedMessage, e.Message);
+                    }
+                }
+
+                var deploymentProperties = new DeploymentProperties(DeploymentMode.Incremental)
+                {
+                    Template = JsonDocument.Parse(template).RootElement,
+                    Parameters = parameters
+                };
+                var input = new DeploymentInput(deploymentProperties)
+                {
+                    Location = location,
+                };
+
+                string deployment = "deployment_" + DateTime.UtcNow.ToString("yyyyMMddHmmffff");
+
+                try
+                {
+                    var deploymentCreateOrUpdateAtScopeOperation = await deploymentCollection.CreateOrUpdateAsync(deployment, input);
+
+                    if (deploymentCreateOrUpdateAtScopeOperation.HasValue &&
+                        deploymentCreateOrUpdateAtScopeOperation.GetRawResponse().Status == 200)
+                    {
+                        return LangServerResources.DeploymentSuccessfulMessage;
+                    }
+                }
+                catch (Exception e)
+                {
+                    var errorMessage = e.Message;
+                    var indexOfHeader = errorMessage.IndexOf("Headers:");
```

Comment:
I feel like we should keep the headers since they may contain useful information for debugging such as correlation IDs.

Reply:
I don't see anything useful in the headers. Here's an example:
Headers:
Cache-Control: no-cache
Pragma: no-cache
x-ms-failure-cause: REDACTED
x-ms-ratelimit-remaining-subscription-writes: REDACTED
x-ms-request-id: 21a7f5ab-e70a-4c28-a412-10afedef71ce
x-ms-correlation-request-id: REDACTED
x-ms-routing-request-id: REDACTED
Strict-Transport-Security: REDACTED
X-Content-Type-Options: REDACTED
Date: Wed, 09 Mar 2022 02:45:04 GMT
Content-Type: application/json; charset=utf-8
Expires: -1
Content-Length: 422

Reply:
Is `x-ms-correlation-request-id` redacted by the SDK? If yes, I'm open to remove the headers.

Reply:
x-ms-correlation-request-id is not redacted by the SDK. Can x-ms-correlation-request-id be used to retrieve deployment information from the portal?

Reply:
Adding the headers back:
https://github.com/Azure/bicep/commit/23eb5e1a1051c92dac87f54f5e135cf4df71f7ad

@ucheNkadiCode, had an offline discussion with @shenglol regarding this and he has a valid point. Sometimes the errors might be deeply nested and if the user wants to reach out to support team, the request id will be useful. Let me know if you have any concerns/questions.

Thanks!

Reply:
Marking this conversation resolved. Let me know if there are any questions. Thanks!